src/controllers/comments_controller.ts

```diff
@@ -1,30 +1,31 @@
-const { handleMongoQueryError } = require("../db");
-const Comment = require("../models/comments_model");
-const Post = require("../models/posts_model");
+import Post from "../models/posts_model";
+import { Request, Response } from "express";
+import { handleMongoQueryError } from "../../db";
+import Comment, { CommentDocument } from "../models/comments_model";
 
-const getComments = async (req, res) => {
+export const getComments = async (req: Request, res: Response): Promise<any> => {
```

Comment:
```suggestion
export const getComments = async (req: Request, res: Response) => {
```

I think that in this case, letting typescript infer the return type is better.
Looks like making this work without any is removing the `return` from the function.
Do it for all the functions